### PR TITLE
Trace sampling + more instrumentation

### PR DESCRIPTION
**Pull request by mystenmark**
- more tracing
- Add --dump-spans
- raise rate limits on ingestion for local cluster
- Trace sampling
  - Trace sample rate can be adjusted via /enable-tracing route

### CODE REVIEW

File: crates/sui-node/src/admin.rs [hunk 1]
```diff
@@ -116,21 +122,50 @@ async fn enable_tracing(
...
-    {
-        Ok(()) => (
-            StatusCode::OK,
-            format!("tracing enabled for {:?}", duration),
-        ),
-        Err(err) => (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()),
+    match state.tracing_handle.update_trace_filter(&filter, duration) {
+        Ok(()) => {
+            response.push(format!("filter set to {:?}", filter));
+            response.push(format!("filter will be reset after {:?}", duration));
```
Comment (longbowlu) [inline crates/sui-node/src/admin.rs:162]: this is probably in the previous PR - where is `reset_tracing` (or the function does reset filters) called?

File: crates/telemetry-subscribers/src/lib.rs [hunk 2]
```diff
@@ -252,7 +255,7 @@ impl TelemetryConfig {
             panic_hook: true,
             crash_on_panic: false,
             prom_registry: None,
-            sample_nth: None,
+            sample_rate: 1.0,
```
Comment (longbowlu) [inline crates/telemetry-subscribers/src/lib.rs:258]: does this set the sample rate to 100% by default?

File: crates/telemetry-subscribers/src/lib.rs [hunk 3]
```diff
@@ -469,46 +471,52 @@ impl TelemetryConfig {
...
-            target_prefix,
+            sample_rate: Arc::new(AtomicF64::new(Self::clamp(sample_rate))),
         }
     }
 
-    fn is_sampled(&self) -> bool {
-        self.counter.fetch_add(1, Ordering::Relaxed) % self.sample_nth == 0
+    fn clamp(sample_rate: f64) -> f64 {
+        // clamp sample rate to between 0.0001 and 1.0
+        sample_rate.max(0.0001).min(1.0)
```
Comment (longbowlu) [inline crates/telemetry-subscribers/src/lib.rs:496]:
```suggestion
        sample_rate.max(1.0).min(0.0001)
```
?

Comment (mystenmark) [reply crates/telemetry-subscribers/src/lib.rs:496]: the original code is correct - min/max is always confusing

File: crates/telemetry-subscribers/src/lib.rs (hunk 2 above)
Comment (mystenmark) [reply crates/telemetry-subscribers/src/lib.rs:258]: yes - we can change this if for some reason its problematic to set the sample rate explicitly in prod

**Pull request merged**